x/nexus/abci.go

```diff
@@ -12,6 +15,53 @@ import (
 func BeginBlocker(_ sdk.Context, _ abci.RequestBeginBlock, _ types.Nexus) {}
 
 // EndBlocker called every block, process inflation, update validator set.
-func EndBlocker(_ sdk.Context, _ abci.RequestEndBlock, _ types.Nexus) []abci.ValidatorUpdate {
+func EndBlocker(ctx sdk.Context, _ abci.RequestEndBlock, n types.Nexus, r types.RewardKeeper, s types.Snapshotter) ([]abci.ValidatorUpdate, error) {
+	if err := checkChainMaintainers(ctx, n, r, s); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func checkChainMaintainers(ctx sdk.Context, n types.Nexus, r types.RewardKeeper, s types.Snapshotter) error {
+	for _, chain := range n.GetChains(ctx) {
+		if !n.IsChainActivated(ctx, chain) {
+			continue
+		}
+
+		rewardPool := r.GetPool(ctx, chain.Name)
+		params := n.GetParams(ctx)
+
+		for _, maintainerState := range n.GetChainMaintainerStates(ctx, chain) {
+			missingVoteCount := maintainerState.MissingVotes.CountTrue(100)
```

Comment:
Shouldn't this pass `window` instead of hard-coding 100?

Reply:
damn good catch